File: angularapp/src/app/services/feedback.service.ts

import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Feedback } from '../models/feedback.model';
import { Observable } from 'rxjs';
import { url } from '../Global/global';

@Injectable({
  providedIn: 'root'
})
export class FeedbackService {

  constructor(private httpClient:HttpClient) { }

  public apiUrl = url.apiUrl+"/api/feedback"


  sendFeedback(feedback:Feedback):Observable<any>
  {
    return this.httpClient.post(this.apiUrl,feedback);
  }

  getAllFeedbacksByUserId(userId:number):Observable<any>
  {
    console.log("get all feedbacks");
    return this.httpClient.get(this.apiUrl+"/user/"+userId);
  }

  deleteFeedback(feedbackId:number):Observable<any>
  {
    return this.httpClient.delete(this.apiUrl+"/"+feedbackId,{ responseType: 'text' });
  }

  getFeedbackById(feedbackId: number): Observable<Feedback> {
    return this.httpClient.get<Feedback>(`${this.apiUrl}/${feedbackId}`);
  }
  
  updateFeedback(feedbackId: number, feedback: Feedback): Observable<any> {
    return this.httpClient.put(`${this.apiUrl}/${feedbackId}`, feedback);
  }  

  getFeedbacks():Observable<any>
  {
    return this.httpClient.get(this.apiUrl);
  }

}

File: springapp/src/main/java/com/examly/springapp/service/FeedbackService.java
package com.examly.springapp.service;

import java.util.List;

import com.examly.springapp.model.Feedback;

public interface FeedbackService {
    public Feedback createFeedback(Feedback feedback);
    public Feedback getFeedbackById(Long id);
    public List<Feedback> getAllFeedbacks();
    public void deleteFeedback(Long id);
    public List<Feedback> getFeedbacksByUserId(Long userId);
    Feedback updateFeedback(Long id, Feedback feedbackDetails);
}
